vpr/src/place/initial_placement.cpp

```diff
@@ -611,50 +822,23 @@ static vtr::vector<ClusterBlockId, t_block_score> assign_block_scores() {
-    auto& cluster_ctx = g_vpr_ctx.clustering();
-
-    auto blocks = cluster_ctx.clb_nlist.blocks();
-
-    std::vector<ClusterBlockId> sorted_blocks(blocks.begin(), blocks.end());
-
-    /*
-     * The criteria considers blocks that belong to a macro or to a floorplan region more difficult to place.
-     * The bigger the macro, and/or the tighter the floorplan constraint, the earlier the block will be in
-     * the list of sorted blocks.
-     * The tiles_outside_of_floorplan_constraints will dominate the criteria, since the number of tiles will
-     * likely be significantly bigger than the macro size. This is okay since the floorplan constraints give
-     * a more accurate picture of how difficult a block is to place.
-     */
-    auto criteria = [block_scores](ClusterBlockId lhs, ClusterBlockId rhs) {
-        int lhs_score = 10 * block_scores[lhs].macro_size + block_scores[lhs].tiles_outside_of_floorplan_constraints + SORT_WEIGHT_PER_FAILED_BLOCK * block_scores[lhs].failed_to_place_in_prev_attempts;
-        int rhs_score = 10 * block_scores[rhs].macro_size + block_scores[rhs].tiles_outside_of_floorplan_constraints + SORT_WEIGHT_PER_FAILED_BLOCK * block_scores[rhs].failed_to_place_in_prev_attempts;
-
-        return lhs_score > rhs_score;
-    };
-
-    std::stable_sort(sorted_blocks.begin(), sorted_blocks.end(), criteria);
-    //print_sorted_blocks(sorted_blocks, block_scores);
-
-    return sorted_blocks;
-}
-
-void print_sorted_blocks(const std::vector<ClusterBlockId>& sorted_blocks, const vtr::vector<ClusterBlockId, t_block_score>& block_scores) {
-    VTR_LOG("\nPrinting sorted blocks: \n");
-    for (unsigned int i = 0; i < sorted_blocks.size(); i++) {
-        VTR_LOG("Block_Id: %zu, Macro size: %d, Num tiles outside floorplan constraints: %d\n", sorted_blocks[i], block_scores[sorted_blocks[i]].macro_size, block_scores[sorted_blocks[i]].tiles_outside_of_floorplan_constraints);
-    }
-}
-
-static void place_all_blocks(enum e_pad_loc_type pad_loc_type, const char* constraints_file) {
+static void place_all_blocks(vtr::vector<ClusterBlockId, t_block_score>& block_scores, enum e_pad_loc_type pad_loc_type, const char* constraints_file) {
     auto& cluster_ctx = g_vpr_ctx.clustering();
     auto& place_ctx = g_vpr_ctx.placement();
     auto& device_ctx = g_vpr_ctx.device();
-
+    auto blocks = cluster_ctx.clb_nlist.blocks();
     int number_of_unplaced_blks_in_curr_itr;
+
     //keep tracks of which block types can not be placed in each iteration
     std::unordered_set<int> unplaced_blk_type_in_curr_itr;
 
+    auto criteria = [&block_scores, &cluster_ctx](const ClusterBlockId& lhs, const ClusterBlockId& rhs) {
+        int lhs_score = 10 * block_scores[lhs].macro_size + block_scores[lhs].number_of_placed_connections + block_scores[lhs].tiles_outside_of_floorplan_constraints + SORT_WEIGHT_PER_FAILED_BLOCK * block_scores[lhs].failed_to_place_in_prev_attempts;
```

Comment:
check magnitude of tiles_outside_of_floorplan_constraints.
Add a comment on relative weighting of the 4 factors here.